src/lib/input-validation/validateDomain.ts

```diff
@@ -9,21 +9,22 @@ import { Enum, ExtendEnum } from "../../types/types";
 import { CommonValidationErrorType } from "./constant/errorType";
 import { validateLength, ValidateLengthErrorType } from "./validateLength";
 
-export const ValidateDomainErrorType = ExtendEnum(
-  ValidateLengthErrorType,
-  CommonValidationErrorType.invalidFormat
-);
+export const ValidateDomainErrorType = ExtendEnum(ValidateLengthErrorType, CommonValidationErrorType.invalidFormat);
 export type ValidateDomainErrorType = Enum<typeof ValidateDomainErrorType>;
 
 export interface ValidateDomainOptions {
   maxLength?: number;
   isRequired?: boolean;
+  isFullDomain?: boolean;
 }
 
 export const validateDomain = (
   value: string,
-  { maxLength = 63, isRequired = true }: ValidateDomainOptions = {}) => {
-  const REGEX = "^\\w[\\w.-]+\\w$";
+  { maxLength = 63, isRequired = true, isFullDomain = false }: ValidateDomainOptions = {}
```

Comment:
can we switch to subDomainOnly or something similar instead?
because it's a validateDomain function, and validating sub domain is extra feature not the other way around

Reply:
updated with `isSubdomainOnly`